examples/ssr/src/instantsearch.js

```diff
@@ -0,0 +1,106 @@
+import instantsearch from 'instantsearch.js/es/index';
+import algoliaHelper from 'algoliasearch-helper';
+
+export const createInstantSearch = ({ searchClient, indexName, options }) => {
+  const search = instantsearch({
+    searchClient,
+    indexName,
+    ...options,
+  });
+
+  search._isSsr = true;
+
+  search.ssr = async params => {
+    search.helper = algoliaHelper(searchClient, indexName, {
+      // parameters set by default
+      highlightPreTag: '__ais-highlight__',
+      highlightPostTag: '__/ais-highlight__',
+      ...params,
+    });
+
+    const { content: lastResults } = await search.helper.searchOnce();
+
+    // TODO: maybe use the life cycle version?
+    search.helper.lastResults = lastResults;
+
+    return {
+      ais: {
+        lastResults: JSON.parse(JSON.stringify(lastResults)),
+      },
+    };
+  };
+
+  search.giveRenderPropsServerOrSomething = widget => {
```

Comment:
used internally only (server): put state from results into widgets, to prevent them rendering empty (start() is not called on server)